mod_amity/amity.py

```diff
@@ -0,0 +1,218 @@
+
+        return staff
+
+    def allocate_person(self, person):
+        """
+        assign space to fellow/staff as requested from the available rooms, as follows
+            Staff: Office
+            Fellow: Office, Living Space(if requested)
+        :param person: created person instance
+        :return: person object with allocations, if any
+        """
+        office = random.choice(self.offices["available"]) if len(self.offices["available"]) > 0 else None
+
+        if office is not None:
+            office.allocate_space(person)
+            person.assign_office(office.name)
+
+        if person.role == Role.FELLOW:
+            living_space = random.choice(self.living_spaces["available"]) \
+                if len(self.living_spaces["available"]) > 0 else None
+            if living_space is not None and person.accommodation == 'Y':
+                living_space.allocate_space(person)
+                person.assign_living_space(living_space.name)
+
+        self.check_person_allocation(person)
+        self.check_room_availability()
+
+        return person
+
+    def get_unallocated_persons(self):
+        """
+        gets persons not fully allocated spaces
+        :return: dict with  fellows and staff
+        """
+        return dict(staff=list(set(self.staff) - set(self.allocated_staff)),
+                    fellows=list(set(self.fellows) - set(self.allocated_fellows)))
+
+    def get_rooms(self, room_name=None):
+        """
+        gets all offices and living spaces in the allocation pool
+        :param room_name: (optional) filter the result by name
+        :return: dict of all living_spaces and offices
+        """
+        offices = self.offices["unavailable"] + self.offices["available"]
+        living_spaces = self.living_spaces["unavailable"] + self.living_spaces["available"]
+        rooms = offices + living_spaces
+        if room_name is not None:
+            for room in rooms:
+                if room.name == room_name:
+                    return room
+        else:
+            return dict(living_spaces=living_spaces, offices=offices)
+
+    def get_staff_id(self):
+        """
+        generate unique ids for staff
+        :return: staff id i.e ST001
+        """
+        staff_id = self.ids["staff"][0] + 1
+        self.ids["staff"][0] = staff_id
+        return "ST{0}".format(str(staff_id).rjust(3, '0'))
+
+    def generate_fellow_id(self):
+        """
+        generate unique ids for fellow
+        :return: staff id i.e FL001
+        """
+        staff_id = self.ids["fellow"][0] + 1
+        self.ids["fellow"][0] = staff_id
+        return "FL{0}".format(str(staff_id).rjust(3, '0'))
+
+    def check_person_allocation(self, person):
+        """
+        checks if fellow/staff is fully allocated and add to allocated list
+        :param person: an instance of fellow or staff
+        :return:
+        """
+        if person.role == Role.STAFF:
+            if person.office is not None:
```

Comment:
The error message is handled by the file handling the docopts commands i.e run.py at the root folder